src/Microdown-Blog/MicSingleSummarizer.class.st

```diff
@@ -2,12 +2,63 @@ Class {
 	#name : 'MicSingleSummarizer',
 	#superclass : 'Object',
 	#instVars : [
-		'maximumWords'
+		'maximumWords',
+		'destinationDirectory',
+		'filePath'
 	],
 	#category : 'Microdown-Blog',
 	#package : 'Microdown-Blog'
 }
 
+{ #category : 'rendering' }
+MicSingleSummarizer >> createHtmlFile: aMicRoot [
+
+	| a q date header |
+	a := (MicHTMLVisitor new visit: aMicRoot) at: 1.
+
+	date := (aMicRoot children at: 1) body at: #date.
+	date := date asDate.
+	header := self firstHeaderBlockOf: aMicRoot.
+
+	q := MicHTMLDocument new.
+	q configuration: a configuration.
+	q configuration document: q.
+	q
+		setCharSetUTF8;
+		destinationPath:
+			destinationDirectory , '\' , date year asString , '\'
+			, date monthName , '\' , date day asString , '\'.
+	filePath := q destinationPath , header text , '.html'.
+	^ q
+		  fileName: header text , '.html';
+		  writeDocument: a contents;
+		  contents
+]
+
+{ #category : 'accessing' }
+MicSingleSummarizer >> destinationDirectory [
+
+	^ destinationDirectory
+]
+
+{ #category : 'accessing' }
+MicSingleSummarizer >> destinationDirectory: aDestination [
+
+	destinationDirectory := aDestination
+]
+
```

Comment:
indeed the summariser as well as all the components date sorter, blogentry...
should all work from a directory to a directory. 

So I asked myself what is filePath vs. destinaryDirectory: why this is not a sourceDirectory